src/fidesops/schemas/saas/saas_config.py

```diff
@@ -93,6 +95,36 @@ def validate_request_for_pagination(cls, values: Dict[str, Any]) -> Dict[str, An
             pagination_strategy.validate_request(values)
         return values
 
+    @root_validator
+    def validate_grouped_inputs(cls, values: Dict[str, Any]) -> Dict[str, Any]:
+        """Validate that grouped_inputs must reference fields from the same collection"""
+        grouped_inputs = values.get("grouped_inputs")
+
+        if grouped_inputs:
+            request_params = values.get("request_params", [])
+            names = [param.name for param in request_params]
+
+            if not all(field in names for field in grouped_inputs):
+                raise ValueError(
+                    "Grouped_input fields must also be declared as request_params."
+                )
+
+            referenced_collections: List[str] = []
+            for param in request_params:
+                if param.name in grouped_inputs:
+                    if not param.references:
+                        raise ValueError(
+                            "Grouped input fields must be on incoming references."
```

Comment:
Note to self, identity data should be allowed here too, this is surfacing in follow-up https://github.com/ethyca/fidesops/issues/291

Reply:
Addressed in https://github.com/ethyca/fidesops/pull/299/commits/fac3b48ac2463a94861e1301afceb78134dff8d8